pkg/runner/runner.go

```diff
@@ -26,81 +29,102 @@ type Runnable interface {
 }
 
 type cmdRunner struct {
-	cmd          exec.Cmd
-	attachStdOut bool
-	attachStdErr bool
+	commandString string
+	cmd           exec.Cmd
+	options       Options
 }
 
 func (c *cmdRunner) String() string {
-	return c.cmd.String()
+	return c.commandString
 }
 
 func (c *cmdRunner) Output() ([]byte, error) {
-	// todo: add logic for sending to stdout as well.
 	copyCMD := c.cmd
-	stdout, err := copyCMD.StdoutPipe()
-	if err != nil {
-		return []byte{}, fmt.Errorf("%w: %v", errGetStdout, err)
+	var stdOutBuf bytes.Buffer
+	var stdErrBuf bytes.Buffer
+
+	var errWriter io.Writer
+	var outWriter io.Writer
+
+	if c.options.SuppressErrOutput {
+		errWriter = io.MultiWriter(&stdErrBuf)
+	} else {
+		errWriter = io.MultiWriter(os.Stderr, &stdErrBuf)
 	}
-	stderr, err := copyCMD.StderrPipe()
-	if err != nil {
-		return []byte{}, fmt.Errorf("%w: %v", errGetStderr, err)
+	if c.options.ShowOutput {
+		outWriter = io.MultiWriter(os.Stdout, &stdOutBuf)
+	} else {
+		outWriter = io.MultiWriter(&stdOutBuf)
 	}
-	rd := bufio.NewReader(stdout)
-	ed := bufio.NewReader(stderr)
+	copyCMD.Stdout = outWriter
+	copyCMD.Stderr = errWriter
 
-	err = copyCMD.Start()
-	if err != nil {
-		return []byte{}, fmt.Errorf("%w: %v", errStartCMDFailure, err)
+	if err := copyCMD.Start(); err != nil {
+		// bad path, binary not executable, &c
+		return nil, err
 	}
 
-	stdOutData := make([]byte, 0)
+	// Make sure to forward signals to the subcommand.
+	cmdChannel := make(chan error) // used for closing the signals forwarder goroutine
+	signalChannel := make(chan os.Signal, 1)
+	signal.Notify(signalChannel, syscall.SIGTERM, syscall.SIGINT)
 	go func() {
 		for {
-			str, err := rd.ReadString('\n')
-			if err != nil {
-				if !errors.Is(err, io.EOF) {
-					fmt.Fprint(os.Stderr, "Runnable read stdout error: ", err)
-				}
+			select {
+			case s := <-signalChannel:
+				copyCMD.Process.Signal(s)
+			case <-cmdChannel:
 				return
 			}
-			if c.attachStdOut {
-				fmt.Fprint(os.Stdout, str)
-			}
-			stdOutData = append(stdOutData, []byte(str)...)
 		}
 	}()
+	defer func() {
+		signal.Stop(signalChannel)
+		signalChannel <- nil
```

Comment:
Do we need to do this? 
The Stop() doc says once it returns, the signal channel is guaranteed to no longer receive any new elements.

Since there are no receivers relying on the channel to close, we don't need to explicitly close it either.

Reply:
Isn't line 88 doing that already?

Reply:
🤦 oops yeah you are right.